src/main/java/bio/terra/service/snapshotbuilder/utils/ConceptChildrenQueryBuilder.java

```diff
@@ -1,47 +1,151 @@
+
+    // concept_ancestor joined on concept.concept_id = ancestor_concept_id
+    var conceptAncestorTablePointer =
+        TablePointer.fromTableName(CONCEPT_ANCESTOR, tableNameGenerator);
+    var conceptAncestorTableVariable =
+        TableVariable.forJoined(conceptAncestorTablePointer, ANCESTOR_CONCEPT_ID, idFieldVariable);
+    var descendantIdFieldVariable =
+        conceptAncestorTableVariable.makeFieldVariable(DESCENDANT_CONCEPT_ID);
+
+    // domain specific occurrence table joined on concept_ancestor.descendant_concept_id =
+    // 'domain'_concept_id
+    var domainOccurrenceTablePointer =
+        TablePointer.fromTableName(domainOption.getTableName(), tableNameGenerator);
+    var domainOccurenceTableVariable =
+        TableVariable.forJoined(
+            domainOccurrenceTablePointer, domainOption.getColumnName(), descendantIdFieldVariable);
+
+    // COUNT(DISTINCT person_id)
+    var countFieldVariable =
+        new FieldVariable(
+            new FieldPointer(domainOccurrenceTablePointer, PERSON_ID, "COUNT"),
+            domainOccurenceTableVariable,
+            "count",
+            true);
+
+    List<SelectExpression> select = List.of(nameFieldVariable, idFieldVariable, countFieldVariable);
+
+    List<TableVariable> tables =
+        List.of(conceptTableVariable, conceptAncestorTableVariable, domainOccurenceTableVariable);
 
-    TablePointer ancestorTablePointer =
-        TablePointer.fromTableName("concept_ancestor", tableNameGenerator);
-    TableVariable ancestorTableVariable = TableVariable.forPrimary(ancestorTablePointer);
-    FieldVariable descendantFieldVariable =
-        ancestorTableVariable.makeFieldVariable("descendant_concept_id");
+    List<FieldVariable> groupBy = List.of(nameFieldVariable, idFieldVariable);
 
-    BinaryFilterVariable whereClause =
+    List<OrderByVariable> orderBy =
+        List.of(new OrderByVariable(nameFieldVariable, OrderByDirection.ASCENDING));
+
+    // ancestorTable is primary table for the subquery
+    var ancestorTablePointer = TablePointer.fromTableName(CONCEPT_ANCESTOR, tableNameGenerator);
+    var ancestorTableVariable = TableVariable.forPrimary(ancestorTablePointer);
+    var descendantFieldVariable = ancestorTableVariable.makeFieldVariable(DESCENDANT_CONCEPT_ID);
+
+    // WHERE c.ancestor_concept_id = conceptId
+    BinaryFilterVariable ancestorClause =
         new BinaryFilterVariable(
-            ancestorTableVariable.makeFieldVariable("ancestor_concept_id"),
+            ancestorTableVariable.makeFieldVariable(ANCESTOR_CONCEPT_ID),
             BinaryFilterVariable.BinaryOperator.EQUALS,
             new Literal(conceptId));
+
+    // WHERE d.descendant_concept_id != conceptId
+    BinaryFilterVariable notSelfClause =
+        new BinaryFilterVariable(
+            ancestorTableVariable.makeFieldVariable(DESCENDANT_CONCEPT_ID),
+            BinaryFilterVariable.BinaryOperator.NOT_EQUALS,
+            new Literal(conceptId));
```

Comment:
I've added some helper methods for these cases, for example:
```suggestion
    BinaryFilterVariable notSelfClause =
        BinaryFilterVariable.notEquals(ancestorTableVariable.makeFieldVariable(DESCENDANT_CONCEPT_ID),
            new Literal(conceptId));
```